frontend/src/components/MapView/Layers/layer-utils.tsx

```diff
@@ -69,11 +73,19 @@ export function getLayerGeometryIcon(layer: LayerType) {
 export const addPopupParams = (
   layer: AdminLevelDataLayerProps | PointDataLayerProps,
   dispatch: Dispatch,
-  evt: any,
+  evt: MapLayerMouseEvent,
   t: i18nTranslator,
   adminLevel: boolean,
 ): void => {
-  const feature = evt.features[0];
+  // TODO: maplibre: fix feature
+  const feature = evt.features?.find(
+    (x: any) => x.layer.id === getLayerMapId(layer.id),
+  ) as any;
+  if (!feature) {
+    return;
+  }
```

Comment:
This block of code appears 3-4 times at least, is it worth extracting into a small function?